### PR TITLE
Move nudging to ddf_common

### PR DESCRIPTION
**Pull request by VimWizardVersace**

### CODE REVIEW

**Review by erickzul — CHANGES_REQUESTED** [review]
Please remove validation_pipeline.ipynb

**Comment by VimWizardVersace** [comment]
> Please remove validation_pipeline.ipynb

done

File: dataset.py [hunk 1]
```diff
@@ -453,3 +454,75 @@ def create_fraudulent_samples(real_samples_data: pd.DataFrame, mean_isoscapes: l
     count += 1
 
   return fake_sample
+
+def valid_in_all_rasters(
```
Comment (erickzul) [inline dataset.py:458]: Please change this to _valid_in_all_rasters to reflect that this should be not used outside of this module (it's a helper method).

File: dataset.py [hunk 2]
```diff
@@ -453,3 +454,75 @@ def create_fraudulent_samples(real_samples_data: pd.DataFrame, mean_isoscapes: l
...
+        df (pd.DataFrame): A dataframe with "lat" and "long" columns.
+        rasters (List[AmazonGeoTiff]): A list of rasters to use to decide whether
+          to nudge coordindates. If a row in the dataframe does not have a value
+          in this raster, nudge it.
+
+    Returns:
+        bool: True if (lat, lon) is valid in all rasters, False if invalid in
+          at least one.
+  '''
+  max_degrees_deviation = 2
```
Comment (erickzul) [inline dataset.py:498]: We want this to be an input parameter to the function. Please consider making a parameter.

File: dataset.py [hunk 3]
```diff
@@ -453,3 +454,75 @@ def create_fraudulent_samples(real_samples_data: pd.DataFrame, mean_isoscapes: l
...
+    are particularly susceptible to being out-of-bounds and will need nudging.
+
+    Args:
+        df (pd.DataFrame): A dataframe with "lat" and "long" columns.
+        rasters (List[AmazonGeoTiff]): A list of rasters to use to decide whether
+          to nudge coordindates. If a row in the dataframe does not have a value
+          in this raster, nudge it.
+
+    Returns:
+        bool: True if (lat, lon) is valid in all rasters, False if invalid in
```
Comment (erickzul) [inline dataset.py:495]: Please change this to:
pd.DataFrame: Returns a dataframe with nudged coordinates.

File: dataset_test.ipynb
Comment (erickzul) [inline dataset_test.ipynb:1]: Is there a way for us to nudge for when the nudging isn't feasible? i.e. the ValueError exception in line 526 in dataset.py

**Review by erickzul — CHANGES_REQUESTED** [review]

File: dataset_test.ipynb
Comment (VimWizardVersace) [reply dataset_test.ipynb:1]: Done

**Review by erickzul — APPROVED** [review]

**Pull request merged**